### PR TITLE
gnupg: take libusb include path from pkg-config

### PR DESCRIPTION
**Pull request by KaiHa**
###### Motivation for this change

This makes smartcards (like the Nitrokey)  usable again.
###### Things done
- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This makes smartcards (like the Nitrokey)  usable again.

### CODE REVIEW

File: pkgs/tools/security/gnupg/fix-libusb-include-path.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/configure
++++ b/configure
+@@ -8872,7 +8872,7 @@
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking libusb include dir" >&5
+ $as_echo_n "checking libusb include dir... " >&6; }
+    usb_incdir_found="no"
+-   for _incdir in "" "/usr/include/libusb-1.0" "/usr/local/include/libusb-1.0"; do
++   for _incdir in "" "/usr/include/libusb-1.0" "/usr/local/include/libusb-1.0" "$($PKG_CONFIG --variable=includedir libusb-1.0)/libusb-1.0"; do
```
Comment (peti) [inline pkgs/tools/security/gnupg/fix-libusb-include-path.patch]: Why didn't you remove those impure paths from the list? Nix builds should never use those locations.

Comment (KaiHa) [inline pkgs/tools/security/gnupg/fix-libusb-include-path.patch]: Done

**Comment by Mic92** [comment]
Should this pull request lead to gpg being linked against libusb?

**Comment by KaiHa** [comment]
My understanding of optional build dependencies is still incomplete, but I believe that libusb was already an optional build dependency before. This change will only fix a glitch that was preventing libusb from being used if it is there.

**Comment by Mic92** [comment]
@KaiHa I don't see how it is being used here. I checked all executable and none is linked against libusb. Or does gpg do dlopen at runtime?

**Comment by KaiHa** [comment]
Sorry for being so unspecific about which part of gpg2 is needing the libusb. It is the _scdaemon_ that is linked against libusb. The scdaemon manages smartcards. I have run `ldd $(dirname $(dirname $(readlink -f $(which gpg2))))/libexec/scdaemon` on it and at least on my system it is linked to _libusb-1.0_.

**Comment by Mic92** [comment]
Also in master now: https://github.com/NixOS/nixpkgs/commit/dace6a176d39c6f3f520337178c6ed62debf50a8

**Comment by KaiHa** [comment]
There was no special reason why I have not removed those paths. I have
simply not thought about it. I will remove the paths and update the
PR.

**Pull request merged**